### ISSUE TITLE
Microsoft.NET.Sdk.Worker projects dont build a container

### ISSUE THREAD

**Issue by CaffeineDaemon** [dotnet/sdk-container-builds#274]
My application contains a project that is setup as worker service with `<Project Sdk="Microsoft.NET.Sdk.Worker">` and is deployed as its own container.

The output for `dotnet publish My.Worker.Project  --os linux --arch x64` stops after the following lines:

```
My.Worker.Project -> {pathToProject}\My.Worker.Project\bin\Debug\net6.0\linux-x64\My.Worker.Project.dll
My.Worker.Project -> {pathToProject}\My.Worker.Project\bin\Debug\net6.0\linux-x64\publish\
```

It does not exit with an error, it just finishes without building or pushing the container, causing errors down the line when either the container image is missing in the registry or using a stale image with same tag.

My other projects with `<Project Sdk="Microsoft.NET.Sdk.Web">` work just as expected, mentioning that the container was build in the `dotnet publish` output:

```
My.Api.Project -> {pathToProject}\My.Api.Project\bin\Debug\net6.0\linux-x64\My.Api.Project.dll
My.Api.Project -> {pathToProject}x\My.Api.Project\bin\Debug\net6.0\linux-x64\publish\
Building image 'my-api-project' with tags v0.10.1-dirty on top of base image mcr.microsoft.com/dotnet/aspnet:6.0
Pushed container 'my-api-project:v0.10.1-dirty' to Docker daemon
```

I tried using `<Project Sdk="Microsoft.NET.Sdk">` for the worker project, but this results in the same behaviour. The same for switching the target framework to `net7.0`.  Both `Sdk` and `Sdk.Worker` are missing the publish targets that are imported for `Sdk.Web` projects, so maybe this is causing the issue.

Changing my worker projekt to `<Project Sdk="Microsoft.NET.Sdk.Web">` works as workaround, but i would prefer if i could use the worker sdk as intended.

**Comment by baronfel** [comment]
This is because the 'worker' templates don't support/import publish profiles in the same way as 'web' templates. I have an issue for this somewhere, but as a workaround you can always call the entrypoint target specificallly as part of your publish. We have this in a few places in our docs, like [Getting Started](https://github.com/dotnet/sdk-container-builds/blob/main/docs/GettingStarted.md).

```
dotnet publish My.Worker.Project --os linux --arch x64 /t:PublishContainer
```

This should be all you need to get going.

**Comment by CaffeineDaemon** [comment]
Thank you @baronfel , this works perfectly for me.

**Comment by baronfel** [comment]
I'm going to close this since we've documented this workaround, but we are aiming to fix this hole for the 7.0.300 series of SDKs, thanks to some information from our friends on the Web Tools team.

**Comment by kaylumah** [comment]
@baronfel will it still be fixed in the 7.0.300 series?

**Comment by baronfel** [comment]
No - we weren't able to get the Publish SDK included for non-Web projects for 7.0.3xx. There are two options:

* Add the Publish SDK (`Microsoft.NET.Sdk.Publish`) to the `Sdk` attribute of a project file: `<Project Sdk="Microsoft.NET.Sdk;Microsoft.NET.Sdk.Publish">` or <Sdk Include="Microsoft.NET.Sdk.Publish" />` and then use the `-p PublishProfile=DefaultContainer` form of the `publish` command, or
* don't add the Publish SDK and continue to use the `/t:PublishContainer` form of the `publish` command for another release cycle.
